### DISCUSSION TITLE
[U-Boot] Does nor flash work now?

### DISCUSSION THREAD

**Discussion by ltrsunny** [linux-chenxing/linux-chenxing.org#66] [General]
mstar_rebase_mainline_20220603
Config is `mstar_infinity2m_som2d01_defconfig` with:
```
CONFIG_SPL_DM_SPI=y
CONFIG_SPL_DM_SPI_FLASH=y
CONFIG_SPL_NOR_SUPPORT=y
CONFIG_SPL_SPI_FLASH_SUPPORT=y
CONFIG_SPL_SPI_LOAD=y
CONFIG_SYS_SPI_U_BOOT_OFFS=0x20000
```
I burned vendor's ipl to 0x0000, uboot's ipl to 0x10000 and u-boot-fit-dtb.bin to 0x20000.

bootlog:
```
delaying for 100
readback: 4277
mplldbg 2
mpll: 0 - ecf70dc9
mpll: 4 - e79ef1f
mpll: 8 - bda1f0bf
mpll: c - 3f7aeadb
mpll: 10 - 9d390161
Trying to boot from SPI
set speed: 1000000
set mode: 0

```

**Comment by ltrsunny** [comment]
mstar_infinity2m_miyoomini_defconfig on this version cannot be compiled.
```
/usr/bin/arm-linux-gnu-ld: board/miyoo/miyoomini/mstarv7.o (symbol from plugin): in function 
embedded_dtb_select:
(.text+0x0): multiple definition of 'mplldbg'; arch/arm/mach-mstar/board.o (symbol from plugin):(.text+0x0): first defined here
/usr/bin/arm-linux-gnu-ld: board/miyoo/miyoomini/mstarv7.o (symbol from plugin): in function embedded_dtb_select':
(.text+0x0): multiple definition of 'mpllregs'; arch/arm/mach-mstar/board.o (symbol from plugin):(.text+0x0): first defined here
collect2: 错误：ld 返回 1

```

How do you make miyoomini bootable?

**Comment by fifteenhex** [comment]
mmm I haven't tried updating u-boot on something with NOR flash for a while. On the miyoomini I used the vendor u-boot as I needed the screen init it has.

**Reply by ltrsunny** [reply]
> mmm I haven't tried updating u-boot on something with NOR flash for a while. On the miyoomini I used the vendor u-boot as I needed the screen init it has.

Well… so can miyoomini's u-boot boot the latest kernel?

**Reply by fifteenhex** [reply]
> > mmm I haven't tried updating u-boot on something with NOR flash for a while. On the miyoomini I used the vendor u-boot as I needed the screen init it has.
> 
> Well… so can miyoomini's u-boot boot the latest kernel?

Yes. You need to use a kernel with the DTB for miyoomini appended.
